### ISSUE TITLE
plugins: rethink instantiation / hydration

### ISSUE THREAD

**Issue by aeneasr** [react-page/react-page#325]
While creating the first templates, I noticed something very cool, layout components could be used as master templates by leveraging the `createInitialChildren` factory, like here:

```jsx
    createInitialChildren: () => ({
      id: uuid(),
      rows: [{
        id: uuid(),
        cells: [{
          id: uuid(),
          layout: { plugin: new LayoutPlugin(dp), state: dp.createInitialState() },
          rows: [{
            id: uuid(),
            cells: [dp.createInitialChildren()]
          }]
        }]
      }, {
        id: uuid(),
        cells: [{
          id: uuid(),
          layout: { plugin: new LayoutPlugin(mp), state: mp.createInitialState() },
          rows: [{
            id: uuid(),
            cells: [mp.createInitialChildren()]
          }]
        }]
      }]
    })
  })
```

As you can see, I create a two layout cells (header and main content area) here. However, I need to wrap the plugin in a factory `LayoutPlugin` because this method is currently taken as-is. Would be great if that would work without it.

The same issue arises from `createInitialChildren`, as this is not possible at the moment

```
  createInitialChildren: () => ({
    id: uuid(),
    rows: [{
      id: uuid(),
      cells: [{
        content: { plugin: defaultPlugin, state: { importFromHtml: '<h1>foo</h1>' } },
        id: uuid(),
      }]
    }]
  })
```

because `plugin.serialize` is not called by the reducer, thus `editorState` is never set, thus slate does not work. ping @inyono

**Comment by arekkas** [comment]
Also it's kinda stupid to call `uuid()` there. Probably should move that somewhere else.

**Comment by macrozone** [comment]
this is partially fixed in 0.14.0  and 0.15.0:

- createInitialChildren can now return a simple config, where uuid is not needed anymore
- defaultPlugin can now be a layout plugin, so you can have prefilled Editors. 

I think there is still a lot of potential here, e.g. we could make it easier to compose plugins out of smaller plugins, so that you can construct complex layouts with just one plugin.
